MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -8,59 +8,114 @@
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Green Knight",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Old",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "The Ice Age Adventures of Buck Wild",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Tesla",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Vivarium",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false)
+//    ]
+
+    private var movies: [MostPopularMovie] = []
     
-    private let questions: [QuizQuestion] = [
-        QuizQuestion(
-            image: "The Godfather",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "The Dark Knight",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "Kill Bill",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "The Avengers",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "Deadpool",
-            text: "Рейтинг этого фильма больше, чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "The Green Knight",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "Old",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "The Ice Age Adventures of Buck Wild",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Tesla",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Vivarium",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false)
-    ]
-    
-    weak var delegate: QuestionFactoryDelegate?
+    private let moviesLoader: MoviesLoading
+    private let delegate: QuestionFactoryDelegate
     
-    init(delegate: QuestionFactoryDelegate?) {
+    init(moviesLoader: MoviesLoading, delegate: QuestionFactoryDelegate) {
+        self.moviesLoader = moviesLoader
         self.delegate = delegate
     }
     
+    func loadData() {
+        moviesLoader.loadMovies {result in
+            DispatchQueue.main.async { [weak self] in
```

Comment:
Старайтесь захватывать слабо ссылку на себя как можно выше. Тут контекст будет захвачен сначала сильно (оно нам не надо), потом будет ослаблен на главной очереди. Порядок не тот, который ожидается
```suggestion
        moviesLoader.loadMovies { [weak self] result in
            DispatchQueue.main.async { [weak self] in
```